File: backend/src/models/SalesOrder.js
const { DataTypes } = require('sequelize');

const SalesOrder = {
  initModel: (sequelize) => {
    const SalesOrder = sequelize.define(
      'SalesOrder',
      {
        id: {
          type: DataTypes.UUID,
          defaultValue: DataTypes.UUIDV4,
          primaryKey: true,
        },
        order_number: {
          type: DataTypes.STRING,
          allowNull: false,
          unique: true,
        },
        customer_id: {
          type: DataTypes.UUID,
          allowNull: false,
        },
        order_date: {
          type: DataTypes.DATE,
          defaultValue: DataTypes.NOW,
        },
        due_date: {
          type: DataTypes.DATE,
          allowNull: true,
        },
        status: {
          type: DataTypes.ENUM('pending', 'confirmed', 'shipped', 'delivered', 'cancelled'),
          defaultValue: 'pending',
        },
        total_amount: {
          type: DataTypes.DECIMAL(15, 2),
          allowNull: false,
        },
        tax_amount: {
          type: DataTypes.DECIMAL(15, 2),
          defaultValue: 0,
        },
        discount_amount: {
          type: DataTypes.DECIMAL(15, 2),
          defaultValue: 0,
        },
        notes: {
          type: DataTypes.TEXT,
          allowNull: true,
        },
        created_at: {
          type: DataTypes.DATE,
          defaultValue: DataTypes.NOW,
        },
        updated_at: {
          type: DataTypes.DATE,
          defaultValue: DataTypes.NOW,
        },
      },
      {
        tableName: 'sales_orders',
        timestamps: false,
      }
    );
    return SalesOrder;
  },
};

module.exports = SalesOrder;

File: sales_warehouse_management_py/app/models/sales_order.py
from sqlalchemy import Column, String, Boolean, DateTime, Enum, Numeric, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime
from app.core.database import Base
import enum

class OrderStatus(str, enum.Enum):
    PENDING = "pending"
    CONFIRMED = "confirmed"
    SHIPPED = "shipped"
    DELIVERED = "delivered"
    CANCELLED = "cancelled"

class SalesOrder(Base):
    __tablename__ = "sales_orders"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    customer_id = Column(UUID(as_uuid=True), nullable=False)  # Foreign key to customers
    order_date = Column(DateTime, default=datetime.utcnow, nullable=False)
    status = Column(Enum(OrderStatus), default=OrderStatus.PENDING, nullable=False)
    total_amount = Column(Numeric(10, 2), nullable=False)
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    def __repr__(self):
        return f"<SalesOrder {self.id} for customer {self.customer_id}>"